cadence/contracts/bridge/FlowEVMBridgeTokenEscrow.cdc

```diff
@@ -0,0 +1,237 @@
+
+import "Burner"
+import "FungibleToken"
+import "NonFungibleToken"
+import "MetadataViews"
+import "ViewResolver"
+import "FlowToken"
+
+import "EVM"
+
+import "FlowEVMBridgeConfig"
+import "FlowEVMBridgeUtils"
+import "CrossVMToken"
+
+/// This escrow contract handles the locking of fungible tokens that are bridged from Cadence to EVM and retrieval of
+/// locked assets in escrow when they are bridged back to Cadence.
+///
+access(all) contract FlowEVMBridgeTokenEscrow {
+
+    /**********************
+            Getters
+    ***********************/
+
+    /// Returns whether the Locker has been initialized for the given fungible token type
+    ///
+    /// @param forType: Type of the locked fungible tokens
+    ///
+    /// @returns true if the Locker has been initialized for the given fungible token type, otherwise false
+    ///
+    access(all) view fun isInitialized(forType: Type): Bool {
+        return self.borrowLocker(forType: forType) != nil
+    }
+
+    /// Returns the balance of locked tokens for the given fungible token type
+    ///
+    /// @param tokenType: Type of the locked fungible tokens
+    ///
+    /// @returns The balance of locked tokens for the given fungible token type or nil if the locker doesn't exist
+    ///
+    access(all) view fun getLockedTokenBalance(tokenType: Type): UFix64? {
+        return self.borrowLocker(forType: tokenType)?.getLockedBalance() ?? nil
+    }
+
+    /// Returns the type of the locked vault for the given fungible token type
+    ///
+    /// @param tokenType: Type of the locked fungible tokens
+    ///
+    /// @returns The type of the locked vault for the given fungible token type or nil if the locker doesn't exist
+    ///
+    access(all) view fun getViews(tokenType: Type): [Type]? {
+        return self.borrowLocker(forType: tokenType)?.getViews() ?? []
+    }
+
+    /// Resolves the requested view type for the given FT type if it is locked and supports the requested view type
+    ///
+    /// @param tokenType: Type of the locked fungible tokens
+    /// @param viewType: Type of the view to resolve
+    ///
+    /// @returns The resolved view as AnyStruct if the vault is locked and the view is supported, otherwise returns nil
+    ///
+    access(all) fun resolveLockedTokenView(tokenType: Type, viewType: Type): AnyStruct? {
+        // The Locker implements Resolver, which has basic resolveView functionality
+        return self.borrowLocker(forType: tokenType)?.resolveView(viewType) ?? nil
+    }
+
+    /**********************
+        Bridge Methods
+    ***********************/
+
+    /// Initializes the Locker for the given fungible token type if it hasn't been initialized yet
+    ///
+    access(account) fun initializeEscrow(
+        with vault: @{FungibleToken.Vault},
```

Comment:
`with vault: @{FungibleToken.Vault}` is the FT vault being used to initialize escrow. FT Escrow needs an empty vault to wrap in a `Locker` so that future bridged funds can be escrowed. Think the parameter naming is alright?